src/nested_pandas/nestedframe/core.py

```diff
@@ -310,3 +315,85 @@ def dropna(
             )
         )
         return new_df
+
+    def reduce(self, func, *args, **kwargs)  -> NestedFrame:
```

Comment:
I'm going back and forth on this name. I like that it matches the idea of getting a lower dimensional result from a potential higher-dimensionality dataset, but am struggling with the idea that nothing about it suggests that this is where you would throw your analysis code at the package... Can always just hammer this in the docs though

Reply:
Yeah very good points. I don't have strong attachments so can always hammer out later.